### ISSUE TITLE
Crash with GIT_OPT_SET_SSL_CIPHERS option

### ISSUE THREAD

**Issue by edo9300** [libgit2/libgit2#5738]
Since the openssl stream implementation checks for this option when it's globally initialized the first time https://github.com/libgit2/libgit2/blob/d185ab24af9498cfce524e53bb82229079ad2af5/src/streams/openssl.c#L240-L243 it has to be set before calling git_libgit2_init, otherwise it will be impossible to change it afterwards. But in the option parsing, it calls ``git__free`` and ``git__strdup`` https://github.com/libgit2/libgit2/blob/d185ab24af9498cfce524e53bb82229079ad2af5/src/libgit2.c#L301-L302 that with this commit https://github.com/libgit2/libgit2/commit/9865cd1696ac4a3f47991a9a1d79b17cef5edc89 were made function pointers that are set after calling git_libgit2_init, and unset before, thus making this option impossible to use, as setting it before git_libgit2_init will make the library crash, and setting it afterwards will make it pointless.

**Comment by ethomson** [comment]
Indeed, that makes sense.  You should make _no_ calls to libgit2 prior to calling `git_libgit2_init`.  We should change this so that setting this option should update the cipher list after-the-fact.
